ush/gfs/task/analysis.py

```diff
@@ -0,0 +1,411 @@
+        # copy full YAML from executable to ROTDIR
+        src = os.path.join(self.datadir, f'{self.taskname}_{self.cdate}.yaml')
+        dest = os.path.join(os.environ['COMOUTaero'], f"{os.environ['APREFIX']}aeroanl.yaml")
+        if os.path.exists(dest):
+            logging.info(f'{dest} already exists, removing it!')
+            os.remove(dest)
+        shutil.copy(src, dest)
+        logging.info(f'Copied YAML file from {src} to {dest}')
+        #---- NOTE below is 'temporary', eventually we will not be using FMS RESTART formatted files
+        #---- all of the rest of this method will need to be changed but requires model and JEDI changes
+        #---- copy RESTART fv_tracer files for future reference
+        cdate_fv3 = dt.datetime.strptime(self.cdate, '%Y%m%d%H').strftime('%Y%m%d.%H%M%S')
+        comin_ges = os.environ['COMIN_GES']
+        # NOTE that while 'chem' is the $componenet, the aerosol fields are with the 'atmos' tracers
+        comin_ges_atm = comin_ges.replace('chem', 'atmos')
+        fms_bkg_file_template = os.path.join(comin_ges_atm, 'RESTART', f'{cdate_fv3}.fv_tracer.res.tile1.nc')
+        for itile in range(1,7):
+            bkg_path = fms_bkg_file_template.replace('tile1', f'tile{itile}')
+            dest = os.path.join(os.environ['COMOUTaero'], f'aeroges.{os.path.basename(bkg_path)}')
+            if os.path.exists(dest):
+                os.remove(dest)
+            logging.info(f'Copying RESTART {bkg_path} to archive aerosol background')
+            shutil.copy(bkg_path, dest)
+
+        #---- add increments to RESTART files
+        self.add_fms_cube_sphere_increments()
+        #---- move increments to ROTDIR
+        fms_inc_file_template = os.path.join(self.datadir, 'anl', f'aeroinc.{cdate_fv3}.fv_tracer.res.tile1.nc')
+        for itile in range(1,7):
+            inc_path = fms_inc_file_template.replace('tile1', f'tile{itile}')
+            dest = os.path.join(os.environ['COMOUTaero'], os.path.basename(inc_path))
+            if os.path.exists(dest):
+                os.remove(dest)
+            logging.info(f'Copying aerosol FMS cube sphere increment to {dest}')
+            shutil.copy(bkg_path, dest)
+
+    def add_fms_cube_sphere_increments(self):
+        import netCDF4 as nc
+        logging.info('Adding increments to RESTART files')
+        # only need the fv_tracer files
+        cdate_fv3 = dt.datetime.strptime(self.cdate, '%Y%m%d%H').strftime('%Y%m%d.%H%M%S')
+        fms_inc_file_template = os.path.join(self.datadir, 'anl', f'aeroinc.{cdate_fv3}.fv_tracer.res.tile1.nc')
+        comin_ges = os.environ['COMIN_GES']
+        # NOTE that while 'chem' is the $componenet, the aerosol fields are with the 'atmos' tracers
+        comin_ges_atm = comin_ges.replace('chem', 'atmos')
+        fms_bkg_file_template = os.path.join(comin_ges_atm, 'RESTART', f'{cdate_fv3}.fv_tracer.res.tile1.nc')
+        incvars = ['dust1', 'dust2', 'dust3', 'dust4', 'dust5',
```

Comment:
Should all of these filenames and variable lists and such be moved to a yaml so we don't have hard-coded values all over the python code?

Reply:
we can go that approach sure, it makes it a bit trickier in terms of pointing to this YAML config file but I don't think that would be too hard.